### ISSUE TITLE
the error of using placeholder in summary.scalar

### ISSUE THREAD

**Issue by pobingwanghai** [tensorflow/tensorflow#8261]
class Model(object):

  def __init__(self,
               images=None,
               actions=None,
               states=None,
               sequence_length=None,
               reuse_scope=None):
    self.prefix = prefix = tf.placeholder(tf.string, [])
    summaries.append(
          tf.summary.scalar(prefix + '_recon_cost' + str(i), recon_cost))

get the error of *** TypeError: expected string or bytes-like object, use the old summary_scalar is ok

**Comment by prb12** [comment]
Please see https://www.tensorflow.org/api_docs/python/tf/summary/scalar
The first argument should be a **Python** string `name` (you are passing a placeholder for a string tensor).
The second arg should be a tensor... I can't tell from your code fragment what the type of `recon_cost` is.

```
>>> import tensorflow as tf
>>> prefix = tf.placeholder(tf.string, [])
>>> print prefix
Tensor("Placeholder:0", shape=(), dtype=string)
>>> print prefix + 'some string'
Tensor("add:0", shape=(), dtype=string)
>>> t = tf.constant(7)
>>> tf.summary.scalar(prefix + 'string', t)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/google/home/pbar/tensorflow.virtualenv/local/lib/python2.7/site-packages/tensorflow/python/summary/summary.py", line 114, in scalar
    name = _clean_tag(name)
  File "/usr/local/google/home/pbar/tensorflow.virtualenv/local/lib/python2.7/site-packages/tensorflow/python/summary/summary.py", line 86, in _clean_tag
    new_name = _INVALID_TAG_CHARACTERS.sub('_', name)
TypeError: expected string or buffer
>>> tf.summary.scalar('string', t)
<tf.Tensor 'string:0' shape=() dtype=string>
>>>
```

This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by pobingwanghai** [comment]
Interesting, in 0.11 the summary_scalar function only can handle a tensor as input name, but now in 1.0,summary.scalar can only handle a python string? That means the use of placeholder as a prefix-name in summary in 0.11 can not be implemented in 1.0 version unless you open a session to get eval(), very interesting change, may I ask is there any alternative way to use placeholder for summary.scalar without the eval() in a session. As the old summary_scalar function give the use of placeholder as a tutorial.

**Comment by prb12** [comment]
@danmane Perhaps you could comment on the summary API changes.  This does seem to be outside the use case for which they were designed (i.e. emitting lots of different summaries with runtime-generated names).
